public/app/plugins/panel/annolist/AnnoListPanel.tsx

```diff
@@ -181,9 +181,22 @@ export class AnnoListPanel extends PureComponent<Props, State> {
   }
 
   onTagClick = (tag: string, remove?: boolean) => {
+    if (!remove && this.state.queryTags.includes(tag)) {
+      return;
+    }
+
     const queryTags = remove ? this.state.queryTags.filter((item) => item !== tag) : [...this.state.queryTags, tag];
 
-    this.setState({ queryTags });
+    // Logic to ensure keyboard focus isn't lost when the currently
+    // focused tag is removed
+    let nextTag: HTMLElement | undefined = undefined;
+    if (remove) {
+      const focusedTag = document.activeElement;
+      nextTag = (focusedTag!.parentElement?.nextElementSibling?.firstElementChild ??
+        focusedTag!.parentElement?.previousElementSibling?.firstElementChild) as HTMLElement;
```

Comment:
ts nits:
 - it looks like we can use optional chaining instead of the typescript "no trust me!" operator. and/or just bail on the ("impossible"?) edgecase of `focusedTag` being null?
 - can we use an `instanceof HTMLElement` check instead of the assertion? We're trying to reduce usages of `as` in the code!

My litmus test for code like this is: would I be comfortable explicitly throwing an exception if my assumptions are invalid, like:

```typescript
if (!focusedTag) throw new Error("Did not expect activeElement be null");
...
if (!(firstElementChild instanceof HTMLElement)) {
  throw new Error("firstElementChild is an unexpected type")
} 
```

If so, then just do that instead.